public/pages/workflows/workflow_list/workflow_list.test.tsx

```diff
@@ -33,5 +64,105 @@ describe('WorkflowList', () => {
   test('renders the page', () => {
     const { getAllByText } = renderWithRouter();
     expect(getAllByText('Manage existing workflows').length).toBeGreaterThan(0);
+    expect(getAllByText('Name').length).toBeGreaterThan(0);
+    expect(getAllByText('Type').length).toBeGreaterThan(0);
+    expect(getAllByText('Last saved').length).toBeGreaterThan(0);
+    expect(getAllByText('Actions').length).toBeGreaterThan(0);
+    expect(getAllByText('workflow_name_0').length).toBeGreaterThan(0);
+  });
+
+  test('sorting functionality', async () => {
+    const { container, getAllByText, queryByText } = renderWithRouter();
+    expect(getAllByText('workflow_name_0').length).toBeGreaterThan(0);
+
+    const sortButtons = container.querySelectorAll(
+      '[data-test-subj="tableHeaderSortButton"]'
+    );
+
+    // Sort workflows list by Name
+    expect(sortButtons[0]).toBeInTheDocument();
+    userEvent.click(sortButtons[0]!);
+    await waitFor(() => {
+      expect(queryByText('workflow_name_19')).toBeInTheDocument();
+      expect(queryByText('workflow_name_0')).toBeNull();
+    });
+    userEvent.click(sortButtons[0]!);
+    await waitFor(() => {
+      expect(queryByText('workflow_name_0')).toBeInTheDocument();
+      expect(queryByText('workflow_name_9')).toBeInTheDocument();
+      expect(queryByText('workflow_name_10')).toBeNull();
+      expect(queryByText('workflow_name_19')).toBeNull();
+    });
+
+    // Sort workflows list by Type
+    expect(sortButtons[1]).toBeInTheDocument();
+    userEvent.click(sortButtons[1]!);
+    await waitFor(() => {
+      expect(getAllByText('Custom').length).toBeGreaterThan(0);
+      expect(queryByText('Unknown')).toBeNull();
+    });
+    userEvent.click(sortButtons[1]!);
+    await waitFor(() => {
+      expect(queryByText('Unknown')).toBeNull();
+      expect(getAllByText('Custom').length).toBeGreaterThan(0);
```

Comment:
@ohltyler, Added  **waitFor** for the assertion instead of userEvent.click after some research. 
- In the above case, It ensures that the component has re-rendered after the click event, and the assertion is made only when the expected text is visible.

- On the other hand, When added to **userEvent.click** , The waitFor is waiting for the click event to happen, but it doesn't wait for the component to re-render after the click event.

Reply:
I see, so it's essentially re-running the assertion until some default timeout, or it passes. Which in our case, should pass, once the click has happened and everything has been re-rendered. makes sense, thanks!